src/bosh-director/lib/bosh/director/jobs/update_stemcell.rb

```diff
@@ -81,41 +82,50 @@ def perform
         stemcell = nil
         cloud_factory.all_configured_clouds.each do |cloud|
           cpi_suffix = " (cpi: #{cloud[:name]})" unless cloud[:name].blank?
-
+          supported = true
```

Comment:
It looks like you could move the `is_supported?` check to this location instead of nesting the `supported` logic throughout this method. If you did an `if !is_supported?` at this location, you could fast fail out of this method. There you could log the cpi format error, raise `StemcellNotSupported` and no other modifications to this method are necessary.